src/builtins.ts

```diff
@@ -113,11 +114,24 @@ function any(x: collections.Iterable): boolean {
 }
 
 function bool(x) {
-    if (x === undefined)
+    
+    if(x instanceof Py_Tuple || x instanceof iterator.XRange || x instanceof pytypes.Py_Str 
+       || x instanceof Py_List || x instanceof Py_Dict){
+        return x.len() !== 0;
+    }
+    if(x instanceof Py_Int || x instanceof Py_Float){
+        return x.toNumber() != 0;
+    }
+    if(x instanceof Py_Long){
+        return !x.eq(Py_Long.fromInt(0));
+    }
+    if(x instanceof Py_Complex){
+        return !(x.real.value == 0 && x.imag.value == 0);
+    }
+    if(x == singletons.None || x == false){
```

Comment:
You should never use `==` if you can help it! It's only useful in a few scenarios, but even then it makes code harder to understand.

```
(function() {
  if (true == 1) {
    alert("double equals performs implicit type conversion; '1' becomes 'true'");
  }
})()
```

`x == singletons.None` should definitely be `x === singletons.None`. `x == false` is only OK if `x` could be a number or a boolean.